core/modules/wpublish/QueriesAndChunks.cc

```diff
@@ -0,0 +1,486 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "wpublish/QueriesAndChunks.h"
+#include "wsched/SchedulerBase.h"
+#include "wsched/ScanScheduler.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.wpublish.QueriesAndChunks");
+}
+
+namespace lsst {
+namespace qserv {
+namespace wpublish {
+
+QueriesAndChunks::QueriesAndChunks(
+    std::chrono::seconds deadAfter,
+    std::chrono::seconds examineAfter, int maxTasksBooted)
+     :_deadAfter{deadAfter}, _examineAfter{examineAfter}, _maxTasksBooted{maxTasksBooted} {
+    auto rDead = [this](){
+        while (_loopRemoval) {
+            removeDead();
+            std::this_thread::sleep_for(_deadAfter);
+        }
+    };
+    std::thread td(rDead);
+    _removalThread = move(td);
```

Comment:
Hmm, I think this another C++ marvel - Koenig Lookup (a.k.a. argument-dependent lookup). `td` is a thread class in `std` namespace so lookup for `move` also searches `std` if it fails to find it in local/global namespaces.